### PR TITLE
Fixed Aggregate Function No Dictionary Column Issue

### PR DESCRIPTION
**Pull request by kumarvishal09**
fixed aggregate function like avg and sum on no dictionary column,

### CODE REVIEW

**Comment by kumarvishal09** [comment]
retest this please

File: core/src/main/java/org/carbondata/query/carbon/aggregator/dimension/impl/VariableLengthDimensionAggregator.java [hunk 1]
```diff
@@ -85,8 +102,24 @@ public VariableLengthDimensionAggregator(DimensionAggregatorInfo dimensionAggera
...
+    if (actualTypeAggregatorIndex.length > 0) {
+      for (int j = 0; j < actualTypeAggregatorIndex.length; j++) {
+        aggeragtor[aggregatorStartIndex + actualTypeAggregatorIndex[j]].agg(dataBasedOnDataType);
+      }
+    }
+    // if sum or avg aggregator is applied then first we need to check whether data type
+    // if data type is string then convert to double data type and then apply aggregate
+    // function
+    if (numberTypeAggregatorIndex.length > 0) {
+      if (dimensionAggeragtorInfo.getDim().getDataType().equals(DataType.STRING)) {
```
Comment (sujith71955) [inline core/src/main/java/org/carbondata/query/carbon/aggregator/dimension/impl/VariableLengthDimensionAggregator.java]: Use constants type check in left side for avoiding unwanted exception to be thrown out not in graceful manner which can deviate the expecteed system behaviour

File: core/src/main/java/org/carbondata/query/carbon/aggregator/dimension/impl/VariableLengthDimensionAggregator.java [hunk 2]
```diff
@@ -46,19 +48,34 @@
...
+   * actual type value like max, min, dictinct count
+   */
+  private int[] actualTypeAggregatorIndex;
+
   public VariableLengthDimensionAggregator(DimensionAggregatorInfo dimensionAggeragtorInfo,
       Object defaultValue, int aggregatorStartIndex, int blockIndex) {
     this.dimensionAggeragtorInfo = dimensionAggeragtorInfo;
     this.defaultValue = defaultValue;
-    this.aggreagtorStartIndex = aggregatorStartIndex;
+    this.aggregatorStartIndex = aggregatorStartIndex;
```
Comment (sujith71955) [inline core/src/main/java/org/carbondata/query/carbon/aggregator/dimension/impl/VariableLengthDimensionAggregator.java]: Please add test cases

Comment (kumarvishal09) [inline core/src/main/java/org/carbondata/query/carbon/aggregator/dimension/impl/VariableLengthDimensionAggregator.java]: ok

File: core/src/main/java/org/carbondata/query/carbon/aggregator/dimension/impl/VariableLengthDimensionAggregator.java (hunk 1 above)
Comment (kumarvishal09) [inline core/src/main/java/org/carbondata/query/carbon/aggregator/dimension/impl/VariableLengthDimensionAggregator.java]: 1. get dimension and get data type cannot be null in any case. i will update the code as its a good practise 
2. ok i will fix

**Comment by kumarvishal09** [comment]
retest this please

**Pull request merged**